File: plugins/scheduler_test.go
package plugins

import (
	"reflect"
	"testing"
)

// qq_no_exec.py  qq_no_shebang.py  qq.py
var in []string

func init() {
	in = []string{
		"./test/qq_no_exec.py",
		"./test/qq_no_shebang.py",
		"./test/qq.py",
	}
}

func TestNoOwnerExecPerm(t *testing.T) {
	expect := []bool{true, true, false}
	for i, v := range in {
		real := noOwnerExecPerm(v)
		if expect[i] != real {
			t.Error("Input value is:", v, "Expected value is:", expect[i], ", Real value is:", real)
		}
	}

}

func TestHasShebang(t *testing.T) {
	expect := []bool{true, false, true}
	for i, v := range in {
		real := hasShebang(v)
		if expect[i] != real {
			t.Error("Input value is:", v, "Expected value is:", expect[i], ", Real value is:", real)
		}
	}
}

func TestGetInterpreterCmd(t *testing.T) {
	expect := []string{
		"/usr/bin/python",
		"-O",
		"./test/qq.py",
	}

	real := getInterpreterCmd(in[2])
	if !reflect.DeepEqual(expect, real) {
		t.Error("Input value is:", in[2], "Expected value is:", expect, "length of value is", len(expect))
		t.Error("Real value is:", real, "length of real value is:", len(real))
	}
}

File: plugins/test/scheduler_test.py
#!/usr/bin/python -O 


def catalan_expr(tups, ops):
    if len(tups)==1:
        yield tups[0]
    else:
        for op in ops:
            for i in range(1, len(tups)):
                for x in catalan_expr(tups[0:i],ops):
                    for y in catalan_expr(tups[i:], ops):
                        yield '(' + op.join((x,y)) + ')'
    
print len(list(catalan_expr('AB','*'))),list(catalan_expr('AB','*'))
print "="*20
print len(list(catalan_expr('ABC','*'))),list(catalan_expr('ABC','*'))
print "="*20
print len(list(catalan_expr('ABCD','*'))),list(catalan_expr('ABCD','*'))
print "="*20
print "="*20
        
print len(list(catalan_expr('AB','/*'))),list(catalan_expr('AB','/*'))
print "="*20
print len(list(catalan_expr('ABC','/*'))),list(catalan_expr('ABC','/*'))
print "="*20
print len(list(catalan_expr('ABCD','/*'))),list(catalan_expr('ABCD','/*'))

